automation-packages/load-testing-okhttp/src/main/java/step/examples/loadtesting/okhttp/OkHttpKeywords.java

```diff
@@ -1,27 +1,170 @@
+    }
+
+    @Keyword(name = "OpenCart Checkout")
+    public void openCartCheckout() {
+        // Step 1
+        assertContains("Guest Shipping", get("https://opencart-prf.exense.ch/index.php?route=checkout/checkout"));
+        // Step 2
+        assertContains("First Name", get("https://opencart-prf.exense.ch/index.php?route=checkout/guest"));
+        assertContains("Ticino", get("https://opencart-prf.exense.ch/index.php?route=checkout/checkout/country&country_id=204"));
+        Map<String, String> data = new HashMap<>(Map.of(
+                "customer_group_id", "1",
+                "firstname", "Gustav",
+                "lastname", "Muster",
+                "email", "gustav@muster.ch",
+                "telephone", "+41777777777",
+                "address_1", "Bahnhofstrasse 1",
+                "city", "Zurich",
+                "postcode", "8001",
+                "country_id", "204",
+                "zone_id", "3120"
+        ));
+        data.putAll(Map.of("company", "", "address_2", ""));
+        assertContains("[]", post("https://opencart-prf.exense.ch/index.php?route=checkout/guest/save", data));
+        // Step 3
+        assertContains("Payment method required", post("https://opencart-prf.exense.ch/index.php?route=checkout/payment_method/save", Map.of("comment", "", "agree", "1")));
+    }
+
+    private String get(String url) {
+        return perform(new Request.Builder().url(url).build());
+    }
+
+    private String post(String url, Map<String, String> params) {
+        FormBody.Builder bodyBuilder = new FormBody.Builder();
+        params.forEach(bodyBuilder::add);
+        Request request = new Request.Builder().url(url).post(bodyBuilder.build()).build();
+        return perform(request);
+    }
+
+    private String perform(Request request) {
+        OkHttpClient client;
+        if (WORKAROUND) {
+            OkHttpClient.Builder clientBuilder = new OkHttpClient.Builder();
+            clientBuilder.setCookieJar$okhttp(new TrivialCookieJar(session));
+            client = clientBuilder.build();
+        } else {
+            client = (OkHttpClient) session.get(OKHTTP_CLIENT);
+        }
 
         try (Response response = client.newCall(request).execute()) {
-            response.body().string();
+            assert response.body() != null;
+            String body = response.body().string();
+            //System.err.println("BODY SIZE: " + body.length());
+            return body;
         } catch (IOException e) {
             throw new RuntimeException(e);
         }
     }
-}
\ No newline at end of file
+
+    private void assertContains(String expected, String body) {
+        if (!body.contains(expected)) {
+            //System.err.println(body);
+            throw new RuntimeException("Response body did not contain expected String: " + expected);
+        }
+    }
+
+    private static class TrivialCookieJar implements CookieJar {
```

Comment:
I assume this is only required for the workaround

Reply:
No, it's required because opencart uses cookies. okhttp only provides an interface, but no actual implementation of a cookie jar. The final implementation without the workaround is 3 lines of actual code...